test/groovy/com/sap/piper/tools/ToolDescriptorTest.groovy

```diff
@@ -0,0 +1,206 @@
+package com.sap.piper.tools
+
+import org.junit.BeforeClass
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.rules.RuleChain
+
+import util.JenkinsLoggingRule
+import util.Rules
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import com.sap.piper.tools.ToolDescriptor
+
+import hudson.AbortException
+
+
+class ToolDescriptorTest extends BasePipelineTest {
+
+    private ExpectedException thrown = new ExpectedException()
+    private JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
+
+    @Rule
+    public RuleChain rules = Rules.getCommonRules(this)
+                .around(thrown)
+                .around(jlr)
+
+    private static tool
+    private static configuration
+
+    private script
+
+
+    @BeforeClass
+    static void init() {
+
+        tool = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', '3.39.10', 'version')
+    }
+
+    @Before
+    void setup() {
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> getNoEnvVars(m) })
+
+        script = loadScript('neoDeploy.groovy').neoDeploy
```

Comment:
Just a question: Below the script loaded here is handed over into ToolDescriptor methods. The point is: ```script``` is a step here. (neoDeploy). I would expect that the reference to the pipeline script is handed over, and not a reference to a step. The pipeline script is normally forwarded to the steps via parameter ```script```. We can see that e.g. in neoDeploy the reference to the step (```this```) is forwarded to e.g. ```getToolExecutable```. Is that really intended? Shouln't it be ```getToolExecutable (script, configuration)``` instead of ```getToolExecutable(this, configuration)```?

Reply:
The script of the step is known, and the script of the pipeline where the step is called is not known, so I think it is good as it is.
Thank you for reviewing the PR.

Reply:
The pipeline script is known. It is handed in into (almost) every step with parameter ```script```.

Reply:
Yes, that's clear, I mean the code of the pipeline script, the implementation, the members that it has , etc, etc. The code/implementation of the step (neoDeploy) is known.
But anyways so far it only access pipeline steps, then so far the implementation of the step is not so important, but is the only benefit that I can see.